test/jdk/javax/swing/JRadioButton/bug4380543.java

```diff
@@ -0,0 +1,160 @@
+                    testObj = new testFrame();
+                    //Adding the Test Frame to handle dispose
+                    PassFailJFrame.addTestFrame(testObj);
+                    PassFailJFrame.positionTestFrame(testObj, PassFailJFrame.Position.HORIZONTAL);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+        passFailJFrame.awaitAndCheck();
+    }
+}
+
+class testFrame extends JFrame implements ActionListener {
+    JPanel buttonsPanel;
+
+    Map<String, String> lookAndFeelMaps = new HashMap<String, String>();
+    public testFrame() throws InterruptedException {
+        initMap();
+        initComponents();
+    }
+
+    public void initMap()
+    {
+        String sLnF;
+        String sMapKey;
+        UIManager.LookAndFeelInfo[] lookAndFeel = UIManager.getInstalledLookAndFeels();
+        for (UIManager.LookAndFeelInfo look : lookAndFeel) {
+
+            sLnF = look.getClassName();
+            sMapKey = sLnF.substring(sLnF.lastIndexOf(".")+1);
+            sMapKey = sMapKey.replaceAll("LookAndFeel","");
+            sMapKey = sMapKey.trim();
+
+            lookAndFeelMaps.put(sMapKey, sLnF);
+        }
+    }
+
+    public void initComponents() throws InterruptedException {
+        JPanel p = new JPanel();
+        buttonsPanel = new JPanel();
+        buttonsPanel.setLayout(new BoxLayout(buttonsPanel, BoxLayout.Y_AXIS));
+
+        JRadioButton rb  = new JRadioButton("JRadioButton");
+        rb.setMargin(new Insets(20, 20, 20, 20));
+        rb.setBackground(Color.green);
+        rb.setAlignmentX(0.5f);
+        buttonsPanel.add(rb);
+
+        JCheckBox cb  = new JCheckBox("JCheckBox");
+        cb.setMargin(new Insets(20, 20, 20, 20));
+        cb.setBackground(Color.yellow);
+        cb.setAlignmentX(0.5f);
+        buttonsPanel.add(cb);
+
+        getContentPane().add(buttonsPanel);
+
+        for (Map.Entry mapElement : lookAndFeelMaps.entrySet()) {
+            String btnName = mapElement.getKey().toString();
+            JButton btn = new JButton(btnName);
+            btn.setActionCommand(btnName);
+            btn.addActionListener(this);
+            p.add(btn);
+        }
+
+        getContentPane().add(p,BorderLayout.SOUTH);
+
+        setSize(500, 300);
+        setVisible(true);
+    }
+
+    private static void setLookAndFeel(String laf) {
+        try {
+            UIManager.setLookAndFeel(laf);
+        } catch (UnsupportedLookAndFeelException ignored) {
+            System.out.println("Unsupported L&F: " + laf);
+        } catch (ClassNotFoundException | InstantiationException
+                | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+    //Changing the Look and Feel on user selection
+    public void actionPerformed(ActionEvent e) {
+        String key = e.getActionCommand();
+        String val = lookAndFeelMaps.get(key);
+
+        setLookAndFeel(val);
```

Comment:
```suggestion
        setLookAndFeel(lookAndFeelMaps.get(e.getActionCommand()));
```
It doesn't make the code harder to read. If you prefer local variables, give them more meaningful names: `val` should rather be `lafClass` and `key` is `lafName` — that's what's stored in the map.